Chat space画面の実装

Koh0214 opened the pull request.

Commit:
css => scss, _reset.scssの読み込み

Commit:
gemのfont-awesome-railsの導入

Commit:
messagesのdb作成

Commit:
messages modelとcontrollerの作成

modelとcontrollerが一緒になってしまいました、すみません。

Commit:
messagesへのroutesの作成

Commit:
messages.scssへresetの読み込み記述

Commit:
application.htmlを修正

中かっこの書き方が間違っていたようなので、ハッシュロケットに戻した。

Commit:
index.htmlのapplication.htmlと重複する部分を削除

yieldの埋め込みも必要ないので消した。

Commit:
hamlを使って外枠を記述

Commit:
messages.scssは要らないので削除

Commit:
headerとleftをとりあえず置いてみた。

Commit:
right側の箱をとりあえず設定。

Commit:
高さと背景色の調整

Commit:
画面の枠を設定

Commit:
cleafix.scssの作成

Commit:
画面の文字情報を書き込むscssの作成とimport

Commit:
right__footer_boxの作成

Commit:
position:fixedが要らないと気づいたので削除

Commit:
html要素の書き込み

editとsendのボックスは後で作る

Commit:
htmlの簡単な中身のところを記述した。

残っていることは
・editとsendのボタン
・font-awesomeを使った部品の表示

Commit:
Editボタンの表示

Commit:
SENDボタンの作成と文字幅の微調整

Commit:
font-awesomeでアイコン表示

Commit:
要らない部分の削除

Koh0214 commented:
![2017-01-07 16 10 40](https://cloud.githubusercontent.com/assets/22772196/21740166/077257be-d4f4-11e6-82a3-d4019b4b0c65.png)

takebnn26 commented on Gemfile:
```diff
@@ -46,3 +46,4 @@ group :development do
 end
 
 gem 'haml-rails'
+gem 'font-awesome-rails'
```
https://gyazo.com/43a7db9e33cb0c587e7dfcea31a7d99b

このようにコメントアウトは消すのはもちろん綺麗にまとめられると良いですね

takebnn26 commented on app/assets/javascripts/messages.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
不要なファイルは消してしまいましょう！

Koh0214 replied:
削除しました！

takebnn26 commented on app/assets/stylesheets/_contents.scss:
```diff
@@ -0,0 +1,103 @@
+header{
```
sassなので入れ子構造で書き直しましょう！
sassならではの書き方（include, mixin, 変数の定義, &, 入れ子構造）をふんだんに使ってください！　

takebnn26 commented on app/assets/stylesheets/_contents.scss:
```diff
@@ -0,0 +1,103 @@
+}
+
+.group_member {
+    color: #999999;
```
2回以上使う色は全て変数に入れて使いまわしましょう！

takebnn26 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,60 @@
+        padding: 0 40px 0 40px;
+    }
+
+    .right__content{
```
&使いましょう！

takebnn26 commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,60 @@
+    height: 100%;
+    width: 300px; 
+    float: left;
+    .left__header{
```
&使いましょう！

takebnn26 commented on app/helpers/messages_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module MessagesHelper
```
今使わない不要なファイルは消してしまいましょう！使うファイルは後から作れば良いので

Koh0214 replied:
削除しました！

takebnn26 commented on app/views/layouts/application.html.haml:
```diff
@@ -2,8 +2,102 @@
     %title ChatSpace
-    = stylesheet_link_tag    'application', media: 'all', 'data: { turbolinks:{ track: true } }'
-    = javascript_include_tag 'application', 'data: {turbolinks: { track: true } }'
+    = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true
```
結論こちらは前の記述のほうが良いです！
というのも、ハッシュロケット=>は使わないのがルールです

takebnn26 commented on app/views/layouts/application.html.haml:
```diff
@@ -2,8 +2,102 @@
   %body
+    %header
+      %p チャットグループが作成されました
+    %div.content
```
%divは全て省略できます！

Koh0214 replied:
%divを全て削除しました！

takebnn26 commented on app/views/layouts/application.html.haml:
```diff
@@ -2,8 +2,102 @@
+      %div.left
+        %div.left__header
+          %div.icons
+            %i.fa.fa-pencil-square-o{"aria-hidden" => "true"}
```
ハッシュロケットの使用は目ておきましょう！

takebnn26 commented on test/controllers/messages_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
こちらも今不要なので消してしまってください
rails gの時に不要なファイルを生成しない設定ができるので調べてやってみましょう！

Koh0214 replied:
削除しました！

takebnn26 commented on test/fixtures/messages.yml:
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
不要なファイルはry

Koh0214 replied:
削除しました！

takebnn26 commented on test/models/message_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
不要なファイルはry

Koh0214 replied:
削除しました！

takebnn26 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,6 @@
+!!!
```
こちらapprication.html.erbの中の = yieldの部分に最終的に挿入されますが、その場合html 要素などがダブってしまいませんか？

Koh0214 replied:
indexの中身を削除しました！

Koh0214 replied:
indexの中身を削除しました！

hironoka replied:
indexの中身を消すのではなく、いまapplication.html.hamlにかかれているchat画面のviewを、index.html.hamlに記述しましょう。yieldについても確認してみてください！

takebnn26 commented on app/views/layouts/application.html.haml:
```diff
@@ -2,8 +2,102 @@
+    = javascript_include_tag 'application', 'data-turbolinks-track' => true
     = csrf_meta_tags
   %body
+    %header
```
どこかのコントローラアクションが呼び出すビューに書き加えてしまいましょう！

hironoka replied:
> どこかのコントローラアクションが呼び出すビューに書き加えてしまいましょう！

↑こちらも修正お願いいたします！
views/messages/index.html.hamlに記述すると良さそうです！

takebnn26 commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root 'messages#index'
+  resources :messages
```
こちらオプションをつけてルーティングをを限定しましょう

Koh0214 replied:
rootをindexのみに限定しました！

Commit:
必要のないファイルの削除

Commit:
rootをindexのみに限定

Commit:
index.htmlがyieldで読み込まれて重複してしまうので削除

Commit:
%divの削除。

hamlでは%divを付ける必要がない。
綺麗に書ける。

Commit:
入れ子構造にした。

Commit:
カラーコードをまとめて、変数で定義。

Commit:
&を使ってクラス名を定義した。

.right{
  .right__header{}
]
みたいのを
.right{
  &__header {}
}
みたいに書ける。

親要素の名前が変数として中に入ってくる感じ。

Commit:
ハッシュロケットをシンボルで書き換え

Commit:
font-awesomeのハッシュロケットを使っていた部分をシンボルへ修正。

hironoka commented on config/routes.rb:
```diff
@@ -1,5 +1,5 @@
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root 'messages#index'
-  resources :messages
+  resources :messages, :only => :index
```
ハッシュロケットは使わずに書いてみてください！
command+Fで「=>」が見つかったらすぐに書き直すようにすると良いと思います！

hironoka replied:
> ハッシュロケットは使わずに書いてみてください！
command+Fで「=>」が見つかったらすぐに書き直すようにすると良いと思います！

こちらも修正お願いいたします！

hironoka commented on app/assets/stylesheets/_contents.scss:
```diff
@@ -1,4 +1,6 @@
-
+$skyblue:   #38AEF0;
```
変数の定義は別のscssファイルを使って切り出すと見やすく、また_contents.scss以外のファイルでも共通して使えるようになります。

stylesheets以下に_valiables.scssなどを作ってまとめてそれをapplication.scssで読み込むようにしてみましょう。

hironoka commented on app/assets/stylesheets/application.scss:
```diff
@@ -2,6 +2,8 @@
 @import "./contents";
 @import "font-awesome";
 
+$rightgrey: #DDDDDD;
```
こちらも変数をまとめてscssファイルを作り、それをapplicatioin.scssで読み込ませるようにしましょう。

hironoka commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,61 @@
+
+$rightgrey: #DDDDDD;
+
+header{
```
header以下、別のscssファイルに切り出して @import を使って読み込ませたほうが、後から確認する時に見やすくなります！

hironoka commented on app/assets/stylesheets/_contents.scss:
```diff
@@ -0,0 +1,105 @@
+.left{
+    color: white;
+    .user_name{
+        font-size: 20px;
```
色だけでなく、フォントサイズも何回か使うものがあれば変数化してしまうと良いです

hironoka commented on app/assets/stylesheets/_contents.scss:
```diff
@@ -0,0 +1,105 @@
+
+    .group_member {
+        color: $rightgrey;
+        font-size: 12px;
```
こちらも何度か出てきているので、変数化すると良いと思います！

Commit:
font-sizeを変数化

Commit:
application.htmlの中身を外に切り出して、import

Commit:
変数を切り出したvariables fileを作成

hironoka commented on app/assets/stylesheets/_contents.scss:
```diff
@@ -1,9 +1,4 @@
-
-$font_big   : 20px;
-$font_small : 12px;
+@import "variables";
```
ここで読み込むと_contents.scssの中でしか使えないので、application.scssで読み込みましょう！

hironoka reviewed (commented):
最後にもう一度レイアウト確認をさせていただきたいので、キャプチャを送ってください！

またSassのファイル構成は明確な決まりはありませんが、最後にもう一度カリキュラムを参考に考えてみてください！
https://gyazo.com/1b72ed0be95e3f1c367bd14f57c901ab

Commit:
変数集のvariablesを大元のapplicationへ記述

Commit:
stylesheetsのmodulesとconfigにファイルを移動、それに伴いパスを書き換えた。

Commit:
ハッシュロケットをシンボルに修正

Koh0214 commented:
![2017-01-07 18 45 46](https://cloud.githubusercontent.com/assets/22772196/21740864/e5b092b0-d509-11e6-810c-e57c445180b2.png)

Commit:
application.htmlの方へ、共通のもののみを残すように編集

koheiiwamura commented on app/assets/stylesheets/modules/_chat_view.scss:
```diff
@@ -0,0 +1,55 @@
+    &__header{
+        background-color: #253141;
+        height: 101px;
+        padding: 0 20px 0 20px;
```
ここ短縮できますね！

koheiiwamura commented on app/assets/stylesheets/modules/_chat_view.scss:
```diff
@@ -0,0 +1,55 @@
+    &__content{
+        background-color: #2f3e51;
+        height: calc(100% - 101px + 70px);
+        padding: 0 20px 0 20px;
```
ここも短縮で！

koheiiwamura commented on app/assets/stylesheets/modules/_chat_view.scss:
```diff
@@ -0,0 +1,55 @@
+    &__header{
+        background-color: white;
+        height: 101px;
+        padding: 0 40px 0 40px;
```
ここもです。

koheiiwamura replied:
padding(上、右、下、左)だけでなく
padding(上下、左右)などもできるので確認してみてください。

Commit:
paddingを短縮

caminogr commented:
rails全体ではインデントは2つなのに対し、stylesheetsファイルのインデントが4つなのが少し気になりますね...!!!
インデントを４つにしている理由って何かありますか？？
特になければ2つにしてしまった方がよさそうです!!

Commit:
インデントの調整

エディタを使い分けていたせいでインデントの数が変わった。

設定変更にて対応済み。

caminogr commented on app/assets/stylesheets/modules/_chat_view.scss:
```diff
@@ -1,55 +1,55 @@
+  width:calc(100vw - 315px);
+  float: left;
+  &__header{
+      background-color: white;
```
このあたりもインデントが４つになっているので２つに修正してください！！！

caminogr reviewed (changes requested):
_reset cssもインデントが４つのままなので修正の方お願いします！！
あともう少しです！！

Commit:
インデントの修正

caminogr commented:
いいですね！
lgtmです
次に進んでください！！
